lib/rule-list.js

```diff
@@ -155,17 +155,95 @@ export const SIMPLIFY_FRACTION = defineRuleString(
 // e.g. 2/-3 -> -2/3
 export const SIMPLIFY_SIGNS = defineRuleString('#a / -#b', '-#a / #b')
 
+// e.g. 2x/3 -> 2/3 x
+export const REWRITE_FRACTIONAL_POLYNOMIAL = defineRuleString(
+    '#a #b / #c', '#a / #c #b', {a: query.isNumber, b: isPolynomialTerm, c: query.isNumber}
+)
 
 // ADDING FRACTIONS
 
+// e.g. 2/5 + 2/5 -> (2 + 2)/5
 export const COMBINE_NUMERATORS =
     defineRuleString('#a_0 / #b + ...', '(#a_0 + ...) / #b')
 
-export const COMMON_DENOMINATOR =
-    defineRuleString(
-        '#a_0 / #b_0 + ...',
-        '(#a_0 * #eval(lcm(#b_0, ...) / #b_0)) / (#b_0 * #eval(lcm(#b_0, ...) / #b_0)) + ...'
-    )
+// Finds common denominator in any scenario
+
+// e.g. 2/6 + 1/4 -> (2 * 2) / (6 * 2) + (1 * 3) / (4 * 3)
+// e.g. 2/(3 + x) + 2/3 -> (2 * 3)/ (3 * (3 + x)) + 2 * (3 + x) / (3 * (3 + x))
+export const COMMON_DENOMINATOR = defineRule(
+    (node) => {
+        if (query.isAdd(node)) {
+            const terms = node.args
+            // checks if at least one of the terms is a fraction
+            const hasFraction = terms.some(term => query.isDiv(term))
+
+            let sameDenom
+            // rule cannot apply if all of the terms are fractions
+            // and they all have the same denominator
+            if (terms.every(term => query.isDiv(term))) {
+                const denom = terms[0].args[1]
+                sameDenom = terms.every(term => print(term.args[1]) == print(denom))
+            }
+
+            return hasFraction && !sameDenom ? {node} : null
+        }
+    },
+
+    (node) => {
+        const terms = node.args
+        // an array storing the index where fraction is negative
+        const negatives = terms.map(term => query.isNeg(term))
+
+        // get numerators and denominators of all fractions
+        let nums = terms.map(term => query.isNeg(term)
+                             ? term.args[0].args[0]
+                             : term.args[0])
+        let denoms = terms.map(term => query.isNeg(term)
+                               ? term.args[0].args[1]
+                               : term.args[1])
+        let newDenoms
+        let newNumerators
+
+        /*
+          If all denoms are numbers, the new denom is
+          the [denom * (LCM / denom)] and the new numerators
+          is [num * (LCM / denom)]
+
+          Else the new denom is the product of all the terms
+          and the new numerator is [num * all other denoms]
+        */
+        if(denoms.every(denom => query.isNumber(denom))) {
+            denoms = denoms.map(denom => query.getValue(denom))
+            const LCM = lcm(...denoms)
+            newDenoms = denoms.map(
+                denom => build.mul(build.number(denom),
+                                   build.number(LCM / denom)))
+            newNumerators = nums.map(
+                (num, i) => build.mul(num,
+                                      build.number(LCM / denoms[i])))
+        } else {
+            newDenoms = build.mul(...denoms)
```

Comment:
This isn't ideal, if all of the denominators are `x` then we'll make this overly complicated.